src/components/common/ModalTrigger.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import { createStyles, withStyles, WithStyles } from '@mui/styles';
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogActions,
+  DialogContent,
+  IconButton,
+  CustomTheme,
+  Typography,
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import LoadingSpinner from 'components/common/LoadingSpinner';
+
+const styles = (theme: CustomTheme) =>
+  createStyles({
+    wrapper: {
+      padding: theme.spacing(4),
+      margin: theme.spacing(4),
+    },
+    dialogTitle: {
+      borderBottom: `1px solid ${theme.palette.divider}`,
+      margin: 0,
+      padding: theme.spacing(2),
+    },
+    closeButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+    dialogContent: {
+      margin: 0,
+      padding: `${theme.spacing(2)} !important`,
+    },
+    dialogInstructions: {
+      margin: 0,
+      padding: `${theme.spacing(2)} !important`,
+      whiteSpace: 'pre-wrap',
+    },
+    dialogActions: {
+      borderTop: `1px solid ${theme.palette.divider}`,
+      margin: 0,
+      padding: theme.spacing(1),
+    },
+    openButton: {
+      width: '100%',
+      border: 0,
+      justifyContent: 'left',
+      textTransform: 'none',
+      paddingTop: theme.spacing(1),
+      paddingBottom: theme.spacing(1),
+      '&:hover': {
+        border: 0,
+      },
+    },
+    openButtonIcon: {
+      marginRight: 5,
+      top: theme.spacing(1),
+    },
+    horizontalModalButton: {
+      fontSize: theme.typography.h6.fontSize,
+      color: theme.palette.primary.light,
+    },
+    overlaySpinner: {
+      opacity: 0.9,
+      position: 'absolute',
+      right: 0,
+      bottom: 0,
+      left: 0,
+      width: 'initial',
+      overflow: 'clip',
+      backgroundColor: theme.palette.common.white,
+      zIndex: 100,
+    },
+  });
+
+interface ModalTriggerProps extends WithStyles<typeof styles> {
+  modalText: string;
+  modalContent: string;
+  modalHeading: string;
+  isLoading: boolean;
+}
+
+export function ModalTrigger(props: ModalTriggerProps) {
```

Comment:
I like the idea of a general purpose modal but I wonder if that makes sense in the TDR UI. There are already a number of modals in TDR but each are done as a separate custom class. It would be more compelling if this PR could refactor an existing model to also use this code, to show that there's value in having a general class for modal displays.